pkg/operator/logging/logging.go

```diff
@@ -132,7 +135,7 @@ func loggerFromFile(ctx context.Context, component string) *zap.SugaredLogger {
 // to use the configured *zap.SugaredLogger from the context
 func ConfigureGlobalLoggers(ctx context.Context) {
 	klog.SetLogger(zapr.NewLogger(logging.FromContext(ctx).Desugar()))
-	ctrl.SetLogger(zapr.NewLogger(logging.FromContext(ctx).Desugar()))
+	logctrl.SetLogger(logzap.New())
```

Comment:
It seems like the logger that we set here shouldn't be a new logger. It should be the same logger that we define on L135 of the `operator.go` file

Reply:
Apologies, @jonathan-innis, I may be missing the context here. Without invoking `SetLogger(...)`, the logs won't appear. Additionally, `ConfigureGlobalLoggers()` seems to be invoked later when configuring the loggers in [operator.go L137](https://github.com/kubernetes-sigs/karpenter/blob/c29c8ff7d520f9b71ccecfff8cc707d80aa467cc/pkg/operator/operator.go#L134)."

Reply:
I think we want to call this `ConfigureGlobalLoggers()` first before anything else now that we are no longer using the knative logger here. Take a look at https://github.com/awslabs/operatorpkg/blob/main/context/context.go#L17 where there is a function that creates the context, including the logger and assigns the zap logger as the controller-runtime logger.

Since we aren't using the knative logger anymore I think we can decouple our need for the knative logger in the `ConfigureGlobalLoggers()` function. Happy to sync over Slack if that doesn't make sense since I realize it's a little hard to explain what I mean here over text.

Reply:
@jonathan-innis thanks. Current version looks as
```Go
func ConfigureGlobalLoggers(ctx context.Context) {
	logctrl.SetLogger(zapr.New())
	klog.SetLogger(zapr.New())
} func ConfigureGlobalLoggers(ctx context.Context) {
	logctrl.SetLogger(zapr.New())
	klog.SetLogger(zapr.New())
} 
```
and here is the log snippet 
```
{"level":"info","ts":"2024-04-29T12:35:42Z","logger":"controller","msg":"webhook disabled"}
{"level":"info","ts":"2024-04-29T12:35:42Z","logger":"controller-runtime.metrics","msg":"Starting metrics server"}
{"level":"info","ts":"2024-04-29T12:35:42Z","logger":"controller-runtime.metrics","msg":"Serving metrics server","bindAddress":":8000","secure":false}
{"level":"info","ts":"2024-04-29T12:35:42Z","logger":"controller","msg":"starting server","kind":"health probe","addr":"[::]:8081"}
{"level":"info","ts":"2024-04-29T12:35:42Z","msg":"attempting to acquire leader lease kube-system/karpenter-leader-election..."}
{"level":"info","ts":"2024-04-29T12:35:59Z","msg":"successfully acquired lease kube-system/karpenter-leader-election"}
```
The pattern for the last two lines here doesn't include the caller. One potential solution is to include the caller's name using `WithName`, but it must be static. However, the current implementation might suffice as is.

Reply:
Thanks. I have reverted and fixed the code accordingly. PTAL.